### DISCUSSION TITLE
Slowing scan

### DISCUSSION THREAD

**Discussion by DanielCastronovo** [aquasecurity/trivy#6912] [Bugs]
### Description

Trivy is very slow than TFsec with same rules.

### Desired Behavior

speed scan

### Actual Behavior

We use Trivy with Atlantis, and for a PR with multiples ressources on TF, the trivy scan is slow

### Reproduction Steps

```bash
1. trivy config --skip-dirs "**/.terraform" --exit-code 0 --severity "CRITICAL" --report summary .
2.
3.
...
```

### Target

Filesystem

### Scanner

Misconfiguration

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
52,10s user 3,32s system 92% cpu 59,629 total
```

### Operating System

Linux

### Version

```bash
0.51.0
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by simar7** [comment]
Can you give us an input that we can use to reproduce this? Without any additional info, it's hard for us to know where the slowdown (if-any) is.

**Reply by DanielCastronovo** [reply]
in my context, this is a big module terraform (submodules of AWS official modules), that create a set of a multiples resources.
So, use Trivy to scan a big project is slower than tfsec, because tfsec only scan terraform code by design.

**Reply by simar7** [reply]
Sure. As you mentioned Trivy scans for other things in addition to Terraform, it makes sense for it to take longer. But like I said earlier, without having a reproducible example it's hard for me to say why it's slower, if at all. Having said that, do you still need help?
